### ISSUE TITLE
No error catching

### ISSUE THREAD

**Issue by Wavfact** [Tealium/integration-angularjs#24]
We noticed a bug with the code..  Granted, it's edge case..

If you are running ghostery and have disabled Tealium, it will error out the page with "TypeError: utag.view is not a function", TypeError: utag.link is not a function" and "TypeError: utag.track is not a function".  

This could probably be resolved with wrapping them in try/catch?

```
var track = function( ev, data ) {
        var data = data || tealiumData.getDataLayer( $location.path() );
        var ev = ev || "view";
        var src = config.script_src;

        if ( typeof utag == "undefined" ) {
          getScript( src, function(){
              try {
                  utag.track(ev, data)
              }
              catch(err){return false;}
          })
        } else {
            try {
                utag.track(ev, data)
            }
            catch(err){return false;}
        }

      };
```

Dennis

**Comment by tygavin** [comment]
At a quick glance, it sounds like you are calling utag.view or utag.link directly in your page?  This module defines a wrapper called tealium.track that should be used instead.  The

` if ( typeof utag == "undefined" ) {`

check is designed to protect against this issue of calling tealium.view too soon (before the utag.js file is loaded.)  As long as this angular integration module is loaded you will have a new function to call  ("tealium.track") for use in your page.  Hopefully ghostery is not blocking the angular module from loading.

That said, we may still need the try/catch inside the getScript callback as you mention.  I'll do a test in with Ghostery on to see if this callback function still executes when they block the js file.

**Comment by Wavfact** [comment]
You are correct.. utag.track is the error..  We are using v1.1.1 as the latest breaks gulp.. :(

I modified our code with it an all was right again..

**Comment by gsurrel-orange** [comment]
> That said, we may still need the try/catch inside the getScript callback as you mention. I'll do a test in with Ghostery on to see if this callback function still executes when they block the js file.

It seems to me that the try/catch inside the getScript callback is necessary when using Ghostery. I can reproduce the problem when configuring Ghostery to block everything.

This is the call stack : 
`VM3765 angular.js:13920 TypeError: utag.track is not a function
    at track (http://localhost:9000/tealium_angular.js:112:16)
    at Object.view (http://localhost:9000/tealium_angular.js:65:9)
    at new <anonymous> (http://localhost:9000/my.controller.js:221:21)
    at Object.instantiate (http://localhost:9000/angular.js:4733:14)
    at $controller (http://localhost:9000/angular.js:10369:28)
    at Object.<anonymous> (http://localhost:9000/angular-ui-router.js:4081:28)
    at http://localhost:9000/angular.js:1247:18
    at invokeLinkFn (http://localhost:9000/angular.js:9934:9)
    at nodeLinkFn (http://localhost:9000/angular.js:9335:11)
    at compositeLinkFn (http://localhost:9000/angular.js:8620:13)`

**Comment by tygavin** [comment]
Because we'll need this fixed for more than just Angular implementations, we've contacted Ghostery.  Right now, they plan to release a fix in the upcoming Ghostery 7.1 release.

**Comment by tygavin** [comment]
I haven't tested, but I noticed Ghostery is on 7.2 now:
https://addons.mozilla.org/en-US/firefox/addon/ghostery/versions/
